CMakeLists.txt

```diff
@@ -415,6 +415,9 @@ OCV_OPTION(WITH_OPENCLAMDFFT "Include AMD OpenCL FFT library support" ON
 OCV_OPTION(WITH_OPENCLAMDBLAS "Include AMD OpenCL BLAS library support" ON
   VISIBLE_IF NOT ANDROID AND NOT IOS AND NOT XROS AND NOT WINRT
   VERIFY HAVE_CLAMDBLAS)
+OCV_OPTION(WITH_CLBLAST "Include CLBlast library support" ON
+  VISIBLE_IF TRUE
+  VERIFY HAVE_CLBLAST)
```

Comment:
> VERIFY

This would break existed build configurations with ENABLE_CONFIG_VERIFICATION

/cc @mshabunin

Reply:
I've checked build with clblast and verification, it works fine:
```
cmake \
        -DCMAKE_INSTALL_PREFIX=install \
        -DWITH_QT=ON \
        -DWITH_1394=OFF \
        -DWITH_JASPER=OFF \
        -DWITH_OPENCLAMDFFT=OFF \
        -DWITH_OPENCLAMDBLAS=OFF \
        -DWITH_LAPACK=OFF \
        -DWITH_CLBLAST=ON \
        -DENABLE_CONFIG_VERIFICATION=ON \
        ../opencv

...

--   OpenCL:                        YES (CLBlast INTELVA)
--     Include path:                /work/opencv/3rdparty/include/opencl/1.2 /usr/include
--     Link libraries:              Dynamic load

...

-- Verifying WITH_CLBLAST=ON => 'HAVE_CLBLAST'=TRUE

...
```

BTW, I observe several identical warnings in matmul.dispatch.cpp (GCC 11, Ubuntu 22):
```
/opencv/modules/core/src/matmul.dispatch.cpp:147:31: warning: type qualifiers ignored on cast result type [-Wignored-qualifiers]
  147 |                               (const cl_mem)A.handle(ACCESS_READ), offsetA, lda,

...

/opencv/modules/core/src/matmul.dispatch.cpp:184:31: warning: type qualifiers ignored on cast result type [-Wignored-qualifiers]
  184 |                               (const cl_mem)B.handle(ACCESS_READ), offsetB, ldb,
      |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
> BTW, I observe several identical warnings in matmul.dispatch.cpp (GCC 11, Ubuntu 22):
> 
> ```
> /opencv/modules/core/src/matmul.dispatch.cpp:147:31: warning: type qualifiers ignored on cast result type [-Wignored-qualifiers]
>   147 |                               (const cl_mem)A.handle(ACCESS_READ), offsetA, lda,
> 
> ...
> 
> /opencv/modules/core/src/matmul.dispatch.cpp:184:31: warning: type qualifiers ignored on cast result type [-Wignored-qualifiers]
>   184 |                               (const cl_mem)B.handle(ACCESS_READ), offsetB, ldb,
>       |                               ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> ```

Warnings are resolved by removing `const`.